components/style/stylist.rs

```diff
@@ -1677,11 +1673,7 @@ impl PerOriginCascadeData {
     }
 
     fn clear(&mut self) {
-        self.element_map = SelectorMap::new();
-        self.pseudos_map = Default::default();
-        SelectorImpl::each_eagerly_cascaded_pseudo_element(|pseudo| {
-            self.pseudos_map.insert(pseudo, SelectorMap::new());
-        });
+        *self = Self::new();
```

Comment:
While we’re touching this, could/should this use `HashMap::clear` to keep and reuse existing heap allocations? Or do we want memory usage to be able to shrink?

r=me either way